deployment/arm/global.json

```diff
@@ -30,6 +39,67 @@
         }
     },
     "resources": [
+        {
+            "type": "microsoft.insights/components",
+            "apiVersion": "2015-05-01",
+            "name": "[parameters('appInsightsName')]",
+            "location": "[resourceGroup().location]",
+            "kind": "web",
+            "properties": {
+                "Application_Type": "web",
+                "Flow_Type": "Bluefield",
+                "Request_Source": "rest"
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-05-01",
+            "name": "[concat('nestedTemplate', copyIndex())]",
```

Comment:
Note, the copy is on a template level as we need to create all the resource within this template multiple times.